source/agora/node/Ledger.d

```diff
@@ -38,11 +44,15 @@ public class Ledger
     /// Temporary storage where transactions are stored until blocks are created.
     private Transaction[] storage;
 
+    /// ledger data dir
+    private const string data_dir;
+
     /// Ctor
-    public this ()
+    public this (const string data_dir)
```

Comment:
I am not sure I understand why `data_dir` is a parameter of the constructor. It does not need to be needed anywhere, as `writeBlockFile` takes `File` arguments, so it does not need to open them.

Reply:
But now you don't need File place because you get it as a parameter.